### PR TITLE
Add Eq1 Ord1 to NonEmptyList LazyNonEmptyList

### LINKED ISSUE #186: Add Eq1 and Ord1 instances to NonEmptyList and LazyNonEmptyList
These instances exist for standard and Lazy lists. I assume we want them in the NonEmpty versions too for completeness.
Related to #183

### PR DESCRIPTION
**Pull request by milesfrain**
**Description of the change**

Adds `Eq1` and `Ord1` instances to `NonEmptyList` and `LazyNonEmptyList`
Fixes #186

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
- [x] Linked any existing issues or proposals that this pull request should close
- [x] Updated or added relevant documentation
- [x] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by milesfrain** [comment]
CI shows that the Lazy version of this change fails to compile, but the non-Lazy version works fine. Don't know why.

File: src/Data/List/Lazy/Types.purs
Comment (milesfrain) [inline src/Data/List/Lazy/Types.purs:215]: Was hoping this shorthand would work, but it does not:
```purs
eq1 = over NonEmptyList eq1
```

**Review by thomashoneyman — APPROVED** [review]

File: src/Data/List/Lazy/Types.purs
Comment (thomashoneyman) [reply src/Data/List/Lazy/Types.purs:215]: I've seen a number of cases where omitting arguments in an instance doesn't work. But I think it's arguably better to include the arguments explicitly in any case.

**Comment by milesfrain** [comment]
Should these changes be tested, and if so, I guess I should also add the same tests to the empty-able lists too?
Also, holding off on editing the changelogs for each of these PR, since that's going to cause a merge conflict after each PR is merged. Will do that right before 0.14

**Review by hdgarrood — APPROVED** [review]
I don’t think these need tests - the type system helps us enough here (via parametricity) that I think we can be confident these are doing the right thing without needing additional tests.

**Comment by hdgarrood** [comment]
Actually let’s include the changelog updated in each PR please; the merge conflicts that would be introduced will all be very easy to fix.

**Comment by hdgarrood** [comment]
Thank you!

File: src/Data/List/Lazy/Types.purs
Comment (milesfrain) [reply src/Data/List/Lazy/Types.purs:215]: I'd argue that the shorthand version is better, since it protects you from making a mistake like this:
```purs
eq1 (NonEmptyList rhs) (NonEmptyList lhs) = eq1 lhs rhs
```

**Pull request merged**
